qiskit_nature/second_q/operators/tensor.py

```diff
@@ -192,6 +201,117 @@ def ndim(self) -> int:
+        :meth:`.SparseLabelOp.from_polynomial_tensor`.
+
+        If no custom template is provided, all instances of ``Tensor`` will resort to the default
+        template. This template depends on the dimension of the wrapped matrix. It will repeat
+        ``{}_{{}}`` for every dimension. This is explained best with an example:
+
+        .. code-block:: python
+
+            print(Tensor(np.eye(2)).label_template)
+            # "{}_{{}} {}_{{}}"
+
+            print(Tensor(np.ones((2, 2, 2, 2)).label_template)
+            # "{}_{{}} {}_{{}} {}_{{}} {}_{{}}"
+
+        The format of this template allows to construct a :class:`.SparseLabelOp` from the
+        ``Tensor``\s stored in a :class:`.PolynomialTensor`. This operation is performed when
+        calling :meth:`.SparseLabelOp.from_polynomial_tensor`. There, the template is processed in
+        two steps:
+
+        1. First, the template is formatted using the key under which the ``Tensor`` object was
+           stored inside of the :class:`.PolynomialTensor` object. For example:
+
+           .. code-block:: python
+
+               poly = PolynomialTensor(
+                  {
+                      "+-": Tensor(np.eye(2)),
+                      "++--": Tensor(np.ones((2, 2, 2, 2))),
+                  }
+              )
+
+               # the label_template will get expanded like so:
+               for key, tensor in poly.items():
+                  sparse_label_template = tensor.label_template.format(*key)
+                  print(key, "->", sparse_label_template)
+
+              # "+-" -> "+_{} -_{}"
+              # "++--" -> "+_{} +_{} -_{} -_{}"
+
+        2. Next, these templates are used to build the actual labels of the :class:`.SparseLabelOp`
+           being constructed. For that, the indices encountered during :meth:`coord_iter` are used
+           for example like so:
+
+           .. code-block:: python
+
+              sparse_label_template = "+_{} -_{}"
+              for value, index in Tensor(np.eye(2)).coord_iter():
+                  sparse_label = sparse_label_template.format(*index)
+                  print(sparse_label, value)
+
+              # "+_0 -_0", 1
+              # "+_0 -_1", 0
+              # "+_1 -_1", 1
+              # "+_1 -_0", 0
+
+        Given that you now understand how the ``label_template`` attribute is being used, this
+        allows you to modify how the ``Tensor`` objects stored inside a :class:`.PolynomialTensor`
+        are processed when they are being translated into a :class:`.SparseLabelOp`.
+
+        Here is a concrete example which enables you to use chemistry-ordered two-body terms:
+
+        .. code-block:: python
+
+           eri_chem = ...  # chemistry-ordered 2-body integrals (a 4-dimensional array)
+           tensor = Tensor(eri_chem)
+           tensor.label_template = "+_{{0}} +_{{2}} -_{{3}} -_{{1}}"
+           poly = PolynomialTensor({"++--": tensor})
```

Comment:
No they don't need to be the same. One could for example map a `+-+-` with a `{0}_{{0}} {2}_{{2}} {3}_{{3}} {1}_{{1}}` template to yield the same result as a `++--` key with the example above.

I wanted to keep the example just a little bit more simpler which is why I hard-coded the `+` and `-` keys in this example.

Do you think I should change this?

Reply:
I think part of it for me was that the first example, after the two step process above, has formatting which will apply only at the second step - since while you could do the template as I showed above, and let the first formatting plug the +/- in, the +/- have instead been manually plugged in there, such that all the first formatting step will do is unescape the curly brackets. Maybe its easier to see overall since its closer to the final formatted form, as you say simpler and for sure its a viable example. Perhaps if it stated there that this is the case for this example. Maybe having the first formatting substitutions, aside from the default template, would be unusual - makes one wonder then if it would not even be more simple for a user to give "+_{0} +_{2} -_{3} -_{1}" where it could bypass the first format pass altogether.

Reply:
I see. I will try to improve the example according to this.

Note, however, that your example `+_{0} +_{2} -_{3} -_{1}` will **not** work because the first formatting is applied unconditionally. I will point this out more explicitly as well.

Reply:
Yes, I realized it would not work, but its simpler to see/deal with (at least for me without the escaped curlies) which I why I ended with `where it could bypass the first format pass altogether.` - but there is no mechanism, as you had it to do that (i.e skip the first format step), and maybe that would make things overall more complicated to deal with versus the escaped curlies.